### PR TITLE
Feature/53 rest api geo zones

### PR DESCRIPTION
**Pull request by onetwo1two**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -3,6 +3,8 @@ def spring_version = '4.2.5.RELEASE'
 def hibernate_version = '5.1.0.Final'
 def postgres_version = '9.4.1208.jre7'
 def hsqldb_version = '2.3.3'
+def jackson_version = '2.7.3'
+def model_mapper_version = '0.7.5'
```
Comment (ITsvetkoFF) [inline build.gradle:7]: it does not deserialize enums
(?)

File: persistence/src/main/java/edu/softserve/zoo/persistence/repository/impl/GeographicalZoneRepository.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package edu.softserve.zoo.persistence.repository.impl;
+
+import edu.softserve.zoo.model.GeographicalZone;
+import org.springframework.stereotype.Component;
+
+
+@Component
```
Comment (ITsvetkoFF) [inline persistence/src/main/java/edu/softserve/zoo/persistence/repository/impl/GeographicalZoneRepository.java]: Repository

File: persistence/src/main/java/edu/softserve/zoo/persistence/specification/impl/GeographicalZoneGetAllSpecification.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package edu.softserve.zoo.persistence.specification.impl;
+
+import edu.softserve.zoo.model.GeographicalZone;
+import edu.softserve.zoo.persistence.specification.HQLSpecification;
+
+public class GeographicalZoneGetAllSpecification implements HQLSpecification<GeographicalZone> {
+    @Override
+    public String query() {
+        return "from GeographicalZone";
```
Comment (ITsvetkoFF) [inline persistence/src/main/java/edu/softserve/zoo/persistence/specification/impl/GeographicalZoneGetAllSpecification.java]: getSimpleName

File: service/src/main/resources/service-ctx.xml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<beans xmlns="http://www.springframework.org/schema/beans"
+       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xmlns:context="http://www.springframework.org/schema/context"
+       xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd http://www.springframework.org/schema/context http://www.springframework.org/schema/context/spring-context.xsd">
+
+    <context:component-scan base-package="edu.softserve.zoo.service.impl"/>
+    <import resource="persistence-ctx.xml"/>
```
Comment (ITsvetkoFF) [inline service/src/main/resources/service-ctx.xml]: ctrl+shift+arrow_up

File: web/src/main/java/edu/softserve/zoo/controller/rest/GeographicalZoneRestController.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+/**
+ * Rest Controller to manage requests, which deal with GeographicalZone
+ *
+ * @author Vadym Holub
+ */
+@RestController
+@RequestMapping("api/v1/geo_zones")
```
Comment (ITsvetkoFF) [inline web/src/main/java/edu/softserve/zoo/controller/rest/GeographicalZoneRestController.java]: to constants

File: service/src/main/java/edu/softserve/zoo/service/GeographicalZoneService.java [hunk 6]
```diff
@@ -0,0 +1,9 @@
+package edu.softserve.zoo.service;
+
+import edu.softserve.zoo.model.GeographicalZone;
+
+import java.util.Collection;
+
+public interface GeographicalZoneService {
```
Comment (ITsvetkoFF) [inline service/src/main/java/edu/softserve/zoo/service/GeographicalZoneService.java]: common interface

File: web/src/main/java/edu/softserve/zoo/controller/rest/GeographicalZoneRestController.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
+package edu.softserve.zoo.controller.rest;
```
Comment (ITsvetkoFF) [inline web/src/main/java/edu/softserve/zoo/controller/rest/GeographicalZoneRestController.java]: Spring Test

File: build.gradle (hunk 1 above)
Comment (ITsvetkoFF) [inline build.gradle:7]: Vadym has checked. It does

File: service/src/test/java/edu/softserve/zoo/service/GeographicalZoneServiceImplTest.java [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+import static org.mockito.Mockito.*;
+
+
+import java.util.Collections;
+
+
+public class GeographicalZoneServiceImplTest extends CrudServiceImplTest<GeographicalZone> {
+
+    @InjectMocks
+    private GeographicalZoneService geographicalZoneService = new GeographicalZoneServiceImpl();
```
Comment (ghost) [inline service/src/test/java/edu/softserve/zoo/service/GeographicalZoneServiceImplTest.java:22]: you should not use mocks for services. wire them instead. you have test context

File: service/src/test/java/edu/softserve/zoo/service/GeographicalZoneServiceImplTest.java [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+    @Mock
+    private GeographicalZoneRepository geographicalZoneRepository;
```
Comment (ghost) [inline service/src/test/java/edu/softserve/zoo/service/GeographicalZoneServiceImplTest.java:25]: wire spring bean

**Pull request closed without merging**